platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -792,7 +825,17 @@ def get_firmware_update_notification(self, image_path):
         return "Immediate power cycle is required to complete {} firmware update".format(self.name)
 
     def install_firmware(self, image_path):
-        return self.__install_firmware(image_path)
+        if MPFAManager.MPFA_EXTENSION in image_path:
+            with MPFAManager(image_path) as mpfa:
+                if not mpfa.get_metadata().has_option('firmware', 'burn'):
+                    raise RuntimeError("Failed to get {} burn firmware".format(self.name))
+
+                burn_firmware = mpfa.get_metadata().get('firmware', 'burn')
+
+                print("INFO: Processing {} burn file: firmware install".format(self.name))
+                self.__install_firmware(os.path.join(mpfa.get_path(), burn_firmware))
```

Comment:
@alexrallen it doesn't comply the API: missing return statement

Reply:
Thanks